internal/controller/nodemaintenance_controller.go

```diff
@@ -353,6 +474,40 @@ func (r *NodeMaintenanceReconciler) handleWaitPodCompletionState(ctx context.Con
 	return res, nil
 }
 
+// updateDrainStatus updates NodeMaintenance drain status in place. returns error if occurred
+func (r *NodeMaintenanceReconciler) updateDrainStatus(ctx context.Context, nm *maintenancev1.NodeMaintenance, drainReq drain.DrainRequest) error {
+	ds, err := drainReq.Status()
+	if err != nil {
+		return fmt.Errorf("failed to update drain status. %w", err)
+	}
+
+	if nm.Status.Drain == nil {
+		// set initial status
+		podsOnNode := &corev1.PodList{}
+		selectorFields := fields.OneTermEqualSelector("spec.nodeName", nm.Spec.NodeName)
+		err = r.Client.List(ctx, podsOnNode, &client.ListOptions{FieldSelector: selectorFields})
+		if err != nil {
+			return fmt.Errorf("failed to list pods. %w", err)
+		}
+
+		nm.Status.Drain = &maintenancev1.DrainStatus{
+			TotalPods:    uint32(len(podsOnNode.Items)),
+			EvictionPods: uint32(len(ds.PodsToDelete)),
+		}
+	}
+
+	removedPods := utils.MaxInt(int(nm.Status.Drain.EvictionPods)-len(ds.PodsToDelete), 0)
+	nm.Status.Drain.DrainProgress = uint32(float32(removedPods) / float32(nm.Status.Drain.EvictionPods) * 100)
```

Comment:
should handle case where EvictionPods is 0

else we divide by zero which is not nice.

Reply:
fixed.